src/Tag/Table.php

```diff
@@ -0,0 +1,171 @@
+        }
+        return $this->caption($caption);
+    }
+
+    /**
+     * @param Colgroup ...$columnGroups One or more column groups ({@see Colgroup}).
+     */
+    public function columnGroups(Colgroup ...$columnGroups): self
+    {
+        $new = clone $this;
+        $new->columnGroups = $columnGroups;
+        return $new;
+    }
+
+    /**
+     * @param Colgroup ...$columnGroups One or more column groups ({@see Colgroup}).
+     */
+    public function addColumnGroups(Colgroup ...$columnGroups): self
+    {
+        $new = clone $this;
+        $new->columnGroups = [...$new->columnGroups, ...$columnGroups];
+        return $new;
+    }
+
+    /**
+     * @param Col ...$columns One or more columns ({@see Col}).
+     */
+    public function columns(Col ...$columns): self
+    {
+        $new = clone $this;
+        $new->columns = $columns;
+        return $new;
+    }
+
+    /**
+     * @param Col ...$columns One or more columns ({@see Col}).
+     */
+    public function addColumns(Col ...$columns): self
+    {
+        $new = clone $this;
+        $new->columns = [...$new->columns, ...$columns];
+        return $new;
+    }
+
+    public function header(?Thead $header): self
+    {
+        $new = clone $this;
+        $new->header = $header;
+        return $new;
+    }
+
+    /**
+     * @param Tbody ...$body One or more body ({@see Tbody}).
+     */
+    public function body(Tbody ...$body): self
+    {
+        $new = clone $this;
+        $new->body = $body;
+        return $new;
+    }
+
+    /**
+     * @param Tbody ...$body One or more body ({@see Tbody}).
+     */
+    public function addBody(Tbody ...$body): self
+    {
+        $new = clone $this;
+        $new->body = [...$new->body, ...$body];
+        return $new;
+    }
+
+    /**
+     * @param Tr ...$rows One or more rows ({@see Tr}).
+     */
+    public function rows(Tr ...$rows): self
+    {
+        $new = clone $this;
+        $new->rows = $rows;
+        return $new;
+    }
+
+    /**
+     * @param Tr ...$rows One or more rows ({@see Tr}).
+     */
+    public function addRows(Tr ...$rows): self
+    {
+        $new = clone $this;
+        $new->rows = [...$new->rows, ...$rows];
+        return $new;
+    }
+
+    public function footer(?Tfoot $footer): self
+    {
+        $new = clone $this;
+        $new->footer = $footer;
+        return $new;
+    }
+
+    protected function generateContent(): string
+    {
+        $items = [];
+
+        if ($this->caption !== null) {
+            $items[] = $this->caption;
+        }
+
+        $items = [
+            ...$items,
+            ...$this->columnGroups,
+            ...$this->columns,
+        ];
+
+        if ($this->header !== null) {
+            $items[] = $this->header;
+        }
+
+        $items = [
+            ...$items,
+            ...$this->body,
+            ...$this->rows,
+        ];
+
+        if ($this->footer !== null) {
+            $items[] = $this->footer;
+        }
```

Comment:
The `tfoot` tag should be placed before `tbody` tag

Reply:
```In this order: optionally a caption element, followed by zero or more colgroup elements, followed optionally by a thead element, followed by either zero or more tbody elements or one or more tr elements, followed optionally by a tfoot element, optionally intermixed with one or more script-supporting elements.```

https://www.w3.org/TR/html52/tabular-data.html#the-table-element